src/science/canopy/cbl_friction_vel.F90

```diff
@@ -11,24 +17,67 @@ MODULE cbl_friction_vel_module
 CONTAINS
 
 SUBROUTINE comp_friction_vel(friction_vel, iter, mp, CVONK, CUMIN, CPI_C,      &
-                             zetar, zref_uv, zref_tq, z0m, ua )
+     zetar, zref_uv, zref_tq, z0m, ua )
+  !*## Purpose
+  !
+  ! This SUBROUTINE evaluates the value of the friction velocity \(u_*\) as used
+  ! during the iteration loop of the Monin-Obuhkov (MO) similarity theory in
+  ! [[define_canopy]]. \(u_*\) is used when quantifying the stability
+  ! parameter `canopy%zetar` in [[update_zetar]], and when evaluating the
+  ! resistance network components `rt0` and `rt1` in [[define_canopy]].
+  ! The friction velocity quantifies the magnitude of turbulent mixing that
+  ! is occurring as well as the flux of momentum from the atmosphere to
+  ! the land.
+  !
+  ! The output `friction_vel` is known as `canopy%us` elsewhere in the code.
+  !
+  ! The basic formula is derived from the modified log-law for the
+  ! vertical profile of wind speed near to the ground namely
+  !
+  ! \( u_{*} = U(z_{ref}) / ( \log [z_{ref}/z_{0m}] -
+  !      \psi_m[\xi z_{ref}/z_{refTq}]
+  !    + \psi_m[\xi z_{0m}/z_{refTq}] ) \)
+  !
+  ! with \(\psi_m\) the integrated similarity function given by [[psim]],
+  ! \(U\) the wind speed at height \(z_{ref}\),
+  ! \(\xi\) the current value of the stability parameter, and \(z_{0m}\) the
+  ! roughness length.
```

Comment:
```suggestion
  ! roughness length.  
```
Add 2 spaces at the end to change line without additional space between paragraphs